RELEASE/scripts/autoscend/auto_bedtime.ash

```diff
@@ -719,6 +719,21 @@ boolean doBedtime()
 	{
 		auto_log_info("Please consider genie wishing for an orcish frat boy spy (You want Frat Warrior Fatigues).", "blue");
 	}
+	
+	if(item_amount($item[Infinite BACON Machine]) > 0 && !get_property("_internetViralVideoBought").to_boolean() && !can_interact())
+	{
+		boolean hasFireworks = (item_amount($item[Clan VIP Lounge Key]) > 0 && get_property("_fireworksShop").to_boolean());
+		boolean hasRetroCape = possessEquipment($item[unwrapped knock-off retro superhero cape]);
+		boolean hasDisintegrate = auto_have_skill($skill[Disintegrate]) && my_maxmp() >= 1.5*mp_cost($skill[Disintegrate]);  //will be limited by current mp, try to gauge if it will be available
+		boolean notNeeded = hasFireworks || hasRetroCape || hasDisintegrate;  //have a common unlimited source of YR, no need to make viral video
```

Comment:
What are your thoughts on changing this is to 
```suggestion
		boolean notNeeded = have_effect($effect[Everything Looks Yellow]) > 0 || canYellowRay();
```
Idea is this will be true if we have used a YR (have effect) or if we can YR now. Figure this is more robust to future YR sources.

Reply:
still considering some relevant changes to canYellowRay(), it does inconsistent preparation and speculation like being only based on current MP

Reply:
just applied the change plus keeping Disintegrate for the MP thing